charts/loki-distributed/templates/_helpers.tpl

```diff
@@ -177,5 +177,30 @@ livenessProbe:
 {{- end -}}
 
 {{- define "loki.config.checksum" -}}
+{{- if .Values.loki.existingSecretForConfig }}
+checksum/config: {{ .Values.loki.externalConfigVersion }}
+{{- else }}
 checksum/config: {{ tpl (mergeOverwrite (tpl .Values.loki.config . | fromYaml) .Values.loki.structuredConfig | toYaml) . | sha256sum }}
 {{- end }}
+{{- end }}
+
+{{/*
+The volume to mount for loki configuration
+*/}}
+{{- define "loki.configVolume" -}}
+{{- if .Values.loki.existingSecretForConfig }}
+{{- if .Values.loki.configAsSecret }}
+secret:
+  secretName: {{ .Values.loki.existingSecretForConfig }}
```

Comment:
💭 I think this maybe a break for some users if previously they were using `existingSecretForConfig` but didn't set `configAsSecret: true`. The break would be with these proposed changes they would need to also set `configAsSecret: true` to ensure the externalConfig is still pulled in from a Secret rather than a ConfigMap.